src/gsi/berror.f90

```diff
@@ -580,17 +579,18 @@ subroutine pcinfo
 !   Only diagonal elements are considered
 
 !   set a coeff. factor for variances of control variables
-    if(diag_precon)then
-      lfact=step_start
-      vprecond=lfact
+    lfact=step_start
+    vprecond=lfact
 
+    if(diag_precon)then
       if(newpc4pred)then
 !       for radiance bias predictor coeff.
         nclen1=nclen-nrclen
         ii=0
         do i=1,jpch_rad
            do j=1,npred
               ii=ii+1
+!             if (ostats(i)>zero) vprecond(nclen1+ii)=vprecond(nclen1+ii)/(one+rstats(j,i)*varprd(ii))
```

Comment:
This is a change that I think should actually be made.  It slightly improves convergence (at least in the first few iterations).  I put it in commented out because it does change the iteration results.

Reply:
OK.  Got it.   This is a feature for possible future use.  If you want to include this change in this PR, feel free to do so.  I trust your judgment.  Does Kristen have any thoughts on this?